integration-tests/step-functions/mocked/audit-event/audit-event.test.ts

```diff
@@ -0,0 +1,883 @@
+                        failedCheckDetails: [
+                          { checkMethod: "data", dataCheck: "record_check" },
+                        ],
+                        txn: "7888835e-07f7-414c-a9a0-62a55e551c2b",
+                        type: "IdentityCheck",
+                      },
+                    ],
+                  },
+                  restricted: {
+                    birthDate: [{ value: "1948-04-23" }],
+                    device_information: { encoded: "test encoded header" },
+                    name: [
+                      {
+                        nameParts: [
+                          { type: "GivenName", value: "Jim" },
+                          { type: "FamilyName", value: "Ferguson" },
+                        ],
+                      },
+                    ],
+                    socialSecurityRecord: [{ personalNumber: "AA000003D" }],
+                  },
+                  timestamp: 1716162264,
+                  user: {
+                    govuk_signin_journey_id:
+                      "252561a2-c6ef-47e7-87ab-93891a2a6a41",
+                    ip_address: "00.100.8.20",
+                    persistent_session_id:
+                      "156714ef-f9df-48c2-ada8-540e7bce44f7",
+                    session_id: "issue-credential-happy-publish",
+                    user_id: "test",
+                  },
+                },
+              }),
+            ])
+          );
+        });
+        it("should exclude extension encoded section when value added is {}", async () => {
+          const input = JSON.stringify({
+            "detail-type": "EVENT_NAME",
+            source: "review-hc.localdev.account.gov.uk",
+            detail: {
+              auditPrefix: "AUDIT_EVENT_PREFIX",
+              deviceInformation: "{}",
+              evidence: [
+                {
+                  failedCheckDetails: [
+                    {
+                      checkMethod: "data",
+                      dataCheck: "record_check",
+                    },
+                  ],
+                  txn: "7888835e-07f7-414c-a9a0-62a55e551c2b",
+                  type: "IdentityCheck",
+                  attemptNum: 2,
+                  ciReasons: [
+                    {
+                      ci: "ci",
+                      reason: "some ci reason",
+                    },
+                  ],
+                },
+              ],
+              user: {
+                govuk_signin_journey_id: "252561a2-c6ef-47e7-87ab-93891a2a6a41",
+                user_id: "test",
+                persistent_session_id: "156714ef-f9df-48c2-ada8-540e7bce44f7",
+                session_id: "issue-credential-happy-publish",
+                ip_address: "00.100.8.20",
+              },
+              userInfoEvent: {
+                Count: 1,
+                Items: [
+                  {
+                    names: {
+                      L: [
```

Comment:
Do you think some of these things could be extracted into variables and reused just to keep this file length a bit more manageable?